oauth.js

```diff
@@ -0,0 +1,91 @@
+var crypto = require('crypto');
+var merge = require('deepmerge');
+
+/**
+ * Creates a superagent plugin to sign API calls using the access token details
+ * compliant with oauth 1, which the API expects
+ * @param {String} accessToken
+ * @param {String} accessTokenSecret
+ * @param {String} apiSecret
+ * @returns {Function}
+ * @see https://github.com/visionmedia/superagent
+ */
+module.exports = function (accessToken, accessTokenSecret, apiSecret) {
+	return function (req) {
+		var queryStringObj = {},
+			queryStrings = req.qsRaw[0],
+			requestParams = req.qs,
+			method = req.method,
+			url = req.url;
+
+		queryStrings.split(',').forEach(function (queryString) {
+			var keyValuePair = queryStrings.split('=');
+
+			queryStringObj[keyValuePair[0]] = keyValuePair[1];
+		});
```

Comment:
So confused, why do we need this? Also, this has to be a typo: `queryStrings` is used in the function body but the argument is `queryString`.